src/ngx_http_datadog_module.cpp

```diff
@@ -461,7 +568,18 @@ static ngx_int_t datadog_init_worker(ngx_cycle_t *cycle) noexcept try {
     return NGX_OK;
   }
 
-  auto maybe_tracer = TracingLibrary::make_tracer(*main_conf);
+  std::shared_ptr<dd::Logger> logger = std::make_shared<NgxLogger>();
+#ifdef WITH_WAF
+  try {
+    security::Library::initialize_security_library(*main_conf);
+  } catch (const std::exception &e) {
+    ngx_log_error(NGX_LOG_ERR, cycle->log, 0,
+                  "Initialising security library failed: %s", e.what());
+    return NGX_ERROR;
```

Comment:
I would argue it should not prevent tracing from working.

Reply:
We have two options:
- Since this is going to be a beta initially, we could fail for now since the original module containing only tracing is still a separate one.
- We could override the relevant `enabled` configuration to `off` and allow tracing to continue.

Either way, in my view this isn't a blocker for now, but we should definitely reevaluate before GA.